pandas/tests/groupby/test_groupby.py

```diff
@@ -2840,3 +2840,15 @@ def test_groupby_index_name_in_index_content(val_in, index, val_out):
     result = series.to_frame().groupby("blah").sum()
     expected = expected.to_frame()
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_on_column_drops_na():
+    # GH 21755
+    df = DataFrame(
+        {"a": [np.nan, "foo", "bar", np.nan, np.nan, "baz"], "b": [0, 1, 2, 3, 4, 5]}
+    )
+    result = df.groupby("a").head(1)
+    expected = DataFrame({"a": ["foo", "bar", "baz"], "b": [1, 2, 5]})
```

Comment:
Can you define the index in here as well? No need to set in an extra line.

Also please merge main

Reply:
I just added a commit which defines the index in the DataFrame constructor, as requested. Also, I have merged the pull request to main.